libraries/common/src/main/java/androidx/media3/common/ForwardingPlayer.java

```diff
@@ -78,7 +81,9 @@ public Looper getApplicationLooper() {
    */
   @Override
   public void addListener(Listener listener) {
-    player.addListener(new ForwardingListener(this, listener));
+    ForwardingListener forwardingListener = new ForwardingListener(this, listener);
+    listeners.put(listener, forwardingListener);
```

Comment:
I wonder if this addition to the bookkeeping set should go after the call to the underlying player (in case it fails for some reason)?